daffodil-tdml-lib/src/test/scala/org/apache/daffodil/tdml/UnitTestTDMLRunner.scala

```diff
@@ -345,6 +345,46 @@ class UnitTestTDMLRunner {
     assertEquals(expected, actual)
   }
 
+  @Test def testCommentBit(): Unit = {
+        val xml = <document bitOrder="LSBFirst">
+                <documentPart type="bits">00000010 //this is a label111</documentPart>
```

Comment:
The syntax highlighting here is a little odd because // is a scala comment. I wonder if that could cause confusion in IDEs? Maybe we should use a comment character that is unique and doesn't necessarily mean anything special in scala as far as syntax highlighting goes?

Also, why did we decide not to use XML comments? Syntax highlighting would just work, and they should be pretty straightforward to remove using normal XML utilities. Is that just too verbose?

Reply:
[DAFFODIL-2369](https://issues.apache.org/jira/browse/DAFFODIL-2369) is worth reading to answer these questions.  // to end of line is only one of two C/C++ comment markers DAFFODIL-2369 suggests - I strongly suggest also implementing the /* */ convention in order to avoid weird syntax highlighting in Scala code like this file.  DAFFODIL-2369 also says breaking up the data into separate documentPart elements in order to use standard XML comments is not effective enough due to the clutter involved.

Reply:
I'm not sure I'm sold on the the clutter argument--it's not great, but it's not awful.

That said, the bug mentions CDATA, which makes me think of what I think is a strong argument against using XML comments, though isn't mentioned in the bug: XML comments aren't treated like comments in CDATA blocks, it's just normal text.

It's not uncommon for TDML tests to used CDATA blocks in the documentPart to ensure IDE's don't rewrap XML and mess up formatting. My main reason for using XML comments is you could just do something like `(testSuite \ "document" \ "documentPart").text` and all the XML comments would be automatically removed. But that's not the case with XML Comments in CDATA, so we would still need special comment removing code anyways. And if we're doing that, we might as well pick something without any potential confusion that comes with XML comments in CDATA, and is maybe a little prettier.

I agree that adding /* */ feels important. Inline comments seem like they could be used a lot.